web/ui/dashboard/src/app/private/pages/project/meta-events/meta-events.component.ts

```diff
@@ -0,0 +1,111 @@
+import { GeneralService } from 'src/app/services/general/general.service';
+import { TagComponent } from 'src/app/components/tag/tag.component';
+import { PrismModule } from 'src/app/private/components/prism/prism.module';
+import { StatusColorModule } from 'src/app/pipes/status-color/status-color.module';
+import { PaginationComponent } from 'src/app/private/components/pagination/pagination.component';
+import { META_EVENT } from 'src/app/models/project.model';
+import { CURSOR, PAGINATION } from 'src/app/models/global.model';
+
+@Component({
+	selector: 'convoy-meta-events',
+	standalone: true,
+	imports: [
+		CommonModule,
+		CardComponent,
+		EmptyStateComponent,
+		ButtonComponent,
+		TooltipComponent,
+		ModalComponent,
+		ModalHeaderComponent,
+		TableLoaderModule,
+		TableCellComponent,
+		TableComponent,
+		TableHeadComponent,
+		TableHeadCellComponent,
+		TableRowComponent,
+		TagComponent,
+		StatusColorModule,
+		PrismModule,
+		PaginationComponent
+	],
+	templateUrl: './meta-events.component.html',
+	styleUrls: ['./meta-events.component.scss']
+})
+export class MetaEventsComponent implements OnInit {
+	metaEventsTableHead: string[] = ['Status', 'Event Types', 'Retries', 'Time', '', ''];
+	showMetaConfig = false;
+	isLoadingMetaEvents = false;
+	isRetryingMetaEvent = false;
+	metaEvents!: { pagination: PAGINATION; content: META_EVENT[] };
+	displayedMetaEvents!: { date: string; content: META_EVENT[] }[];
+	selectedMetaEvent: any;
+
+	constructor(public privateService: PrivateService, private generalService: GeneralService, private metaEventsService: MetaEventsService) {}
+
+	ngOnInit(): void {
+		this.getMetaEvents();
+	}
+
+	get isMetaEventEnabled(): Boolean {
+		const isMetaEventEnabled = localStorage.getItem('IS_META_EVENTS_ENABLED');
+		if (isMetaEventEnabled) return JSON.parse(isMetaEventEnabled);
+		return false;
+	}
+
+	toggleMetaConfig(event: any) {
+		const isConfigfureMetaEventsChecked = event.target.checked;
+		this.showMetaConfig = isConfigfureMetaEventsChecked;
+	}
+
+	async getMetaEvents(requestDetails?: CURSOR) {
+		this.isLoadingMetaEvents = true;
+		try {
+			const response = await this.metaEventsService.getMetaEvents(requestDetails);
+			this.metaEvents = response.data;
+			if (this.metaEvents?.content?.length) this.selectedMetaEvent = this.metaEvents?.content[0];
+			this.displayedMetaEvents = await this.generalService.setContentDisplayed(response.data.content);
+			this.isLoadingMetaEvents = false;
+		} catch {
+			this.isLoadingMetaEvents = false;
+		}
+	}
+
+	async retryMetaEvent(metaEventId: string) {
+		this.isRetryingMetaEvent = true;
+		try {
+			const response = await this.metaEventsService.retryMetaEvent(metaEventId);
+			this.isRetryingMetaEvent = false;
+			this.generalService.showNotification({ message: response.message, style: 'success' });
+			this.getMetaEvents();
+		} catch {
+			this.isRetryingMetaEvent = false;
+		}
+	}
+
+	getCodeSnippetString(type: 'req_header' | 'res_header' | 'res_body') {
```

Comment:
@Oluwadaminiola We have this method somewhere else yea?

Can we put it in a service or helper instead?